src/gui/mainwindow.cpp

```diff
@@ -2106,18 +2105,23 @@ QIcon MainWindow::getSystrayIcon() const
 #endif // Q_OS_MACOS
 
 #if defined(Q_OS_WIN) || defined(Q_OS_MACOS)
-void MainWindow::checkProgramUpdate()
+void MainWindow::checkProgramUpdate(const bool invokedByUser)
 {
-    m_programUpdateTimer->stop(); // If the user had clicked the menu item
+    if (m_programUpdateTimer)
+        m_programUpdateTimer->stop();
+
     m_ui->actionCheckForUpdates->setEnabled(false);
     m_ui->actionCheckForUpdates->setText(tr("Checking for Updates..."));
     m_ui->actionCheckForUpdates->setToolTip(tr("Already checking for program updates in the background"));
-    bool invokedByUser = m_ui->actionCheckForUpdates == qobject_cast<QAction *>(sender());
-    ProgramUpdater *updater = new ProgramUpdater(this, invokedByUser);
-    connect(updater, &ProgramUpdater::updateCheckFinished, this, &MainWindow::handleUpdateCheckFinished);
+
+    auto *updater = new ProgramUpdater(this);
+    connect(updater, &ProgramUpdater::updateCheckFinished
+        , this, [this, invokedByUser, updater](const bool updateAvailable, const QString &newVersion)
```

Comment:
Well, now I see both `invokedByUser` and `programUpdater` instance are handled at caller side so I'm satisfied with this aspect of changes.